presto-main/src/main/java/io/prestosql/metadata/Metadata.java

```diff
@@ -319,6 +324,16 @@
      */
     void finishDelete(Session session, TableHandle tableHandle, Collection<Slice> fragments);
 
+    /**
+     * Begin update query
+     */
+    TableHandle beginUpdate(Session session, TableHandle tableHandle, List<ColumnHandle> updatedColumns, List<Integer> updateExpressionChannelNumbers);
```

Comment:
The `updateExpressionChannelNumbers` give the channel numbers of the projected update SET expressions in the Page supplied to `UpdatablePageSource.updateRows()`.

I've done this kind of thing differently for SQL Merge - - I packaged up the merge SET expressions in a RowType, so I can guarantee the order.

This will take some thought - - I'll commit changes for your other comments and come back to this.

Reply:
As commented below, I got rid of updateExpressionChannelNumbers.  Now it only shows up as an argument to UpdatablePageSource.updateRows().